controllers/raycluster_controller.go

```diff
@@ -304,6 +316,15 @@ func (r *RayClusterReconciler) reconcileAutoscaler(ctx context.Context, rc *dcv1
 	return nil
 }
 
+func (r *RayClusterReconciler) initializeStatus(ctx context.Context, rc *dcv1alpha1.RayCluster) error {
```

Comment:
I don't think so because `RayCluster` and `SparkCluster` types are technically unrelated. I know I can change this, but it's probably too much work to remove this little duplication (plus, this framework has been abandoned by Sonny anyway).